### DISCUSSION TITLE
`error.module.reasons` is missing

### DISCUSSION THREAD

**Discussion by vjpr** [webpack/webpack#15792] [Q&A]
When I get an error in webpack, I want to see the chain of requests that caused it to be called. But I can't find `reasons` when an error happened. 

Is it not included until the bundle is successfully built?

Is there a way to get the chain of requests that led to a "module not found" error?

**Comment by vankop** [comment]
> Is there a way to get the chain of requests that led to a "module not found" error?

no, we don't log this

**Reply by vjpr** [reply]
Any tips on how to get this info? Can I get access to the dep graph in the `stats` object? Or some kind of partial dep graph?

**Reply by vankop** [reply]
```js
compiler => {
   compiler.hooks.done.tap("my-plugin", stats => {
      // find your module in stats.compilation.modules
      const allIncomingConnections = stats.compilation.moduleGraph.getIncomingConnections(module);
   });
}
```
use this as a plugin

**Reply by vjpr** [reply]
Does any code already exist to search for all paths from entry point to a module?

**Reply by vankop** [reply]
not sure what you mean.. you can traverse module graph to the entry point ( using code above )

**Reply by vjpr** [reply]
Just thought maybe there was a function already implemented to do such a thing. But easy enough to write myself.

**Reply by vjpr** [reply]
QQ: Is there any reason there can be a `ModuleGraphConnection` with no `originModule`?

As I understand it, a connection is the graph edge, `conn.module` -- "requested by" --> `conn.originModule`.

I'm guessing the `ModuleGraphConnection` doesn't just model the connection between `module` and `originModule`...

**Reply by vankop** [reply]
entrypoint connection I think should be without `originModule`
